src/aggregates/views.ts

```diff
@@ -1,23 +1,77 @@
 import { UnsequencedVideoEvent, VideoEvent, VideoEventsBucketModel, VideoEventType } from '../models/VideoEvent'
 import { EntityViewsInfo } from '../entities/EntityViewsInfo'
+import { mapPeriods } from '../resolvers/viewsInfo'
+import { differenceInCalendarDays } from 'date-fns'
 
 type VideoEventsAggregationResult = {
   events?: VideoEvent[]
 }[]
 
+type TimePeriodEventsData = {
+  sevenDays: Partial<UnsequencedVideoEvent>[]
+  thirtyDays: Partial<UnsequencedVideoEvent>[]
+}
+
+type TimePeriodViews = {
+  sevenDays: EntityViewsInfo[]
+  thirtyDays: EntityViewsInfo[]
+}
+
 export class ViewsAggregate {
   private videoViewsMap: Record<string, number> = {}
   private channelViewsMap: Record<string, number> = {}
   private categoryViewsMap: Record<string, number> = {}
+  private timePeriodEvents: TimePeriodEventsData = {
+    sevenDays: [],
+    thirtyDays: [],
+  }
+
+  private timePeriodVideoViews: TimePeriodViews = {
+    sevenDays: [],
+    thirtyDays: [],
+  }
+
+  private timePeriodChannelViews: TimePeriodViews = {
+    sevenDays: [],
+    thirtyDays: [],
+  }
+
+  private timePeriodCategoryViews: TimePeriodViews = {
+    sevenDays: [],
+    thirtyDays: [],
+  }
```

Comment:
Why are those arrays holding events? I think we could keep just one list of events (`timePeriodEvents`) and for those just use `Record<string, number>` to map id => views?

Reply:
There is `timePeriodEvents` in line 24. Those arrays holding views for each `videos`, `channels` and `categories`

Reply:
Yeah I'm aware we already have `timePeriodEvents`. I suggest that `timePeriodVideoViews.sevenDays` doesn't have to be an array, it could very well be a mapping instead. So instead of `[{videoId: 1}, {videoId:1}, {videoId:5}]` it would hold `{1: 2, 5: 1}`

Reply:
I'm thinking if this is actually a good idea and what is the advantage of mapping? This way we will have to build array of views on each query. In my solution we can send prepared array of views, without any additional operations. WDYT?